tests/main/preseed-snapd-snap/task.yaml

```
@@ -0,0 +1,105 @@
+summary: Check that preseeding of current ubuntu cloud image works with snapd
+  and core18 works.
+description: |
+  This test checks that preseeding of Ubuntu cloud images with snap-preseed
+  command works, up to the point where the image is ready to be booted.
+  The test assumes cloud image with a core and lxd snaps in its seeds/.
+
+systems: [ubuntu-20.04-*]
+
+environment:
+  IMAGE_MOUNTPOINT: /mnt/cloudimg
+
+prepare: |
+  # the get_image_url_for_nested_vm is a convenient helper that returns
+  # a cloud image url matching current $SPREAD_SYSTEM.
+  #shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+  wget "$(get_image_url_for_nested_vm)" -O cloudimg.img
+  mkdir -p "$IMAGE_MOUNTPOINT"
+
+restore: |
+  # any of the restore commands can fail depending on where execute part stopped,
+  # account for that with ||true.
+  umount_ubuntu_image "$IMAGE_MOUNTPOINT" || true
+
+execute: |
+  setup_preseeding_core18() {
+    local IMAGE_MOUNTPOINT=$1
+    local SEEDS_DIR
+    SEEDS_DIR="$IMAGE_MOUNTPOINT/var/lib/snapd/seed/"
+    rm -f "$SEEDS_DIR"/snaps/*.snap
+    rm -f "$SEEDS_DIR"/seed.yaml
+
+    snap download --edge snapd
+    snap download --edge core18
+
+    CORE=$(ls core18*.snap)
+    SNAPD=$(ls snapd*.snap)
+    sed seed.yaml.in  -E -e 's/@CORE@/'"$CORE"'/' -e 's/@SNAPD@/'"$SNAPD"'/' > "$SEEDS_DIR"/seed.yaml
```

I think the quoting can be simpler here:

```suggestion
    sed seed.yaml.in  -E -e "s/@CORE@/$CORE/" -e "s/@SNAPD@/$SNAPD/" > "$SEEDS_DIR"/seed.yaml
```

---

Indeed, thanks!